__tests__/repositorio/lance.test.js

```diff
@@ -0,0 +1,76 @@
+import { obtemLancesDoLeilao, adicionaLance } from "../../src/repositorio/lance";
+import apiLeiloes from "../../src/servicos/apiLeiloes";
+
+const mockLances = [
+  {
+    id: 1,
+    valor: 10,
+  },
+  {
+    id: 2,
+    valor: 20,
+  },
+];
+
+const mockRequisicao = (retorno) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve({
+        data: retorno
+      });
+    }, 100);
+  });
+};
+
+jest.mock("../../src/servicos/apiLeiloes");
+
+describe("repositorio/lance", () => {
+  beforeEach(() => {
+    apiLeiloes.get.mockClear();
+    apiLeiloes.post.mockClear();
+  });
+
+  describe("obtemLancesDoLeilao", () => {
+    it("deve retornar um array de lances", async () => {
+      apiLeiloes.get.mockImplementation(() => mockRequisicao(mockLances));
+
+      const lances = await obtemLancesDoLeilao(1);
+
+      expect(apiLeiloes.get).toHaveBeenCalledWith('/lances?leilaoId=1&_sort=valor&_order=desc');
+      expect(apiLeiloes.get).toHaveBeenCalledTimes(1);
+      expect(lances).toEqual(mockLances);
+    });
+
+    it("deve retornar um array vazio caso erro na API", async () => {
+      apiLeiloes.get.mockImplementation(() => mockRequisicaoErro());
+
+      const lances = await obtemLancesDoLeilao(1);
+
+      expect(apiLeiloes.get).toHaveBeenCalledWith('/lances?leilaoId=1&_sort=valor&_order=desc');
+      expect(apiLeiloes.get).toHaveBeenCalledTimes(1);
+      expect(lances).toEqual([]);
+    });
+  });
+
+  describe("adicionaLance", () => {
+    it("deve retornar true caso a requisição funcione", async () => {
+      apiLeiloes.post.mockImplementation(() => mockRequisicao());
+
+      const sucesso = await adicionaLance(mockLances[0]);
+
+      expect(apiLeiloes.post).toHaveBeenCalledWith('/lances', mockLances[0]);
+      expect(apiLeiloes.post).toHaveBeenCalledTimes(1);
+      expect(sucesso).toBe(true);
+    });
+
+    it("deve retornar false caso erro na API", async () => {
```

Comment:
Sugestão de frase: "deve retornar false caso dê erro na requisição". Erro na API me parece algo como se a API não estivesse funcionando, mas se esse for o caso pode deixar como está.

Reply:
No caso seria se a api está offline, mas como estou mocando, o que da erro é a requisição, já não tem ver com API mesmo.
[Mudei aqui](https://github.com/alura-cursos/react-native-criando-testes-para-sua-aplicacao/commit/5853e5687e182c0b712c01c64fc602415e062548)